### PR TITLE
Add maxissues threshold support for CI/CD rule enforcement

### PR DESCRIPTION
**Pull request by B1naryVision**
Currently, asist fails if any issues are detected (equivalent to a hard cap of 0). This change allows users to define a custom limit per rule.

💡 Why is this needed?
On large or legacy repositories, "perfect" compliance is often impossible to achieve overnight. By allowing a maxIssues threshold, teams can:

Grandfather in existing issues: Set the cap to the current issue count to prevent new regressions.

Incremental Improvement: Gradually lower the cap over time as technical debt is cleared.

CI Adoption: Enable the analyzer in CI/CD pipelines without immediately breaking the build for existing debt.

🛠 Proposed Changes
Added a maxIssues configuration option per rule.

Default Behavior: If no cap is defined, it defaults to 0, maintaining the current strict enforcement behavior.

Logic: The build fails only if actualIssues > maxIssues.

**Why only CI/CD?**: It only exits with error codes if it's in CI/CD mode, otherwise it simply prints out the issues.
Perhaps we would also like to Print a warning outside of CI/CD? Would that mess with baseline or anything though?

🧪 WIP Notes (Seeking Early Input)
I am raising this as a Draft PR to get feedback on the implementation logic before I finalize the test suite. Specifically:

Does the flag naming (maxIssues) align with the project's naming conventions?
Here's some other options I considered.
- issuethreshold
- maxallowedissues
- cicdmaxissues
- failon

Are there any edge cases in the threshold logic I should account for before I write the unit tests?

I added a printing log in CI/CD mode, is this a **breaking change** as it now no longer ONLY prints the issues?

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -341,15 +362,14 @@ A few differences can be observed compared to regular scans:
 - Each issue is assigned a `FindingID`, which is effectively a hash that uniquely identifies the finding based on the rule, the finding location, and if it's a false positive or not.
 - A few more properties are added, such as if the rule is custom, or if the finding is marked as false positive or not.
 
-
-### 🫣 .gitignore and .forceignore files
+## 🫣 .gitignore and .forceignore files
```
Comment (B1naryVision) [inline docs/index.md]: I did some nitpicking, a H1 -> H3.

Arguably this should have stayed a H3 and CI/CD should be a H2 but...
Happy to change it up 😅

File: docs/index.md
Comment (B1naryVision) [inline docs/index.md]: Future enhancement I might do, is make the index/readme copy from a central file with these details so I don't need to copy-paste across except for the headers/footers. Maybe it does and I just didn't see it?

**Review by gkumarcertinia — DISMISSED** [review]

**Review by jwoodwardcertinia — DISMISSED** [review]
LGTM

**Review by ajaincertinia — COMMENTED** [review]
Hi @B1naryVision, we had internal discussion regarding the `--list-rule` case, and decided that `maxissues` should be renamed to `cicdmaxissues` (as per your earlier suggestion), and `--list-rule` should print this property.

File: config/config.go [hunk 2]
```diff
@@ -129,6 +129,25 @@ func (c *Config) GetCICDRuleIds() []rules.RuleID {
 	return ruleIds
 }
 
+/**
+ * Returns the max allowed issues for a specific rule ID.
+ * Returns ruleoverrides.<rule>.cicdmaxissues if set, otherwise 0.
+ */
+func (c *Config) GetRuleMaxIssues(ruleId rules.RuleID) int {
```
[the remark this answers is not in the document]
Comment (ajaincertinia) [reply config/config.go]: It's okay to me.

**Review by ajaincertinia — DISMISSED** [review]

**Review by gkumarcertinia — DISMISSED** [review]

**Review by gkumarcertinia — APPROVED** [review]

**Review by ajaincertinia — APPROVED** [review]
